mixer/pkg/perf/controller_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perf
+
+import (
+	"testing"
+)
+
+func TestControllerBasic(t *testing.T) {
+	c, err := newController()
+	if err != nil {
+		t.Fatalf("Error creating controller: %v", err)
+	}
+
+	s, err := NewClientServer(c.location())
```

Comment:
It will be awesome if the controller takes a config of all the Clients and just inits it for me and waits etc...
From users point of view, this model has two ways to trigger runClient, one by invoking runClients on the controller and other by directly executing the `Run` method on the NewClientServer object.

Reply:
Controller is meant to be internal only. The only reason it is "public" is because the RPC framework requires it. The only supported way to execute things is to call the "Run*" methods.